src/codegen/tests/typesystem_test.rs

```diff
@@ -254,3 +254,60 @@ fn int_bigger_than_byte_promoted_on_compare_statement() {
 
     insta::assert_snapshot!(result);
 }
+
```

Comment:
these tests are fine  ... 
I would like some additional tests that specifically test the annotation like the tests in: src/resolver/tests/resolve_expressions_tests.rs

they build a certain situation an tests on the annotation on certain  AST elements.

I think you can write good tests that also document the behavior pretty nicely.

llike GIVEN a variadic function without type declarations
WHEN i call it with ints and floats
THEN the resolver should promote them to LINT/LREAL